cmd/xl-v1-metadata.go

```diff
@@ -86,9 +86,11 @@ func (c *ChecksumInfo) UnmarshalJSON(data []byte) error {
 	if err != nil {
 		return err
 	}
-	c.Name = info.Name
 	c.Algorithm = BitrotAlgorithmFromString(info.Algorithm)
 	c.Hash = sum
+	if _, err = fmt.Sscanf(info.Name, "part.%d", &c.PartNumber); err != nil {
```

Comment:
Not a huge fan of this. Are we sure we are backwards compatible, ie. will every version of this be formatted this way?

Changing it is a good idea and I can't really think is something better.

Reply:
Yes we are backwards compatible, I will confirm once again.